pkg/ebpf/perf.go

```diff
@@ -50,31 +51,33 @@ func NewPerfHandler(dataChannelSize int) *PerfHandler {
 		RecordGetter: func() *perf.Record {
 			return recordPool.Get().(*perf.Record)
 		},
+		closed: make(chan struct{}),
 	}
 }
 
 // LostHandler is the callback intended to be used when configuring PerfMapOptions
 func (c *PerfHandler) LostHandler(CPU int, lostCount uint64, perfMap *manager.PerfMap, manager *manager.Manager) {
-	if c.closed {
+	select {
+	case <-c.closed:
 		return
+	case c.LostChannel <- lostCount:
+	default:
```

Comment:
This doesn't actually protect from the panic because of the random selection mechanic of `select`. See https://go.dev/play/p/ep9CoLNHpB8

Reply:
We are not closing the `LostChannel` channel anymore.

Reply:
But I could move the check for `c.closed` into its own select, so we don't randomly write to other channels if we `Stop` has already been called.

Reply:
That's a problem for users of `PerfHandler` then, yeah? I think the expectation is the channels will close when data stops.